panic in `broker.(*Broker).Subscribe`

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xb530ad]

goroutine 2930507 [running]:
code.vegaprotocol.io/data-node/broker.(*Broker).Subscribe(0x0, 0x153f378, 0xc01b4cbb00, 0x1)
	/jenkins/workspace/vegaprotocol_data-node_develop/data-node/broker/broker.go:206 +0x2d
code.vegaprotocol.io/data-node/subscribers.(*Service).ObserveEvents(0xc000251170, 0x1535fd8, 0xc01b4e4400, 0x3, 0xc02177b3d8, 0x1, 0x1, 0x0, 0xc01b35ca20, 0x1, ...)
	/jenkins/workspace/vegaprotocol_data-node_develop/data-node/subscribers/service.go:34 +0x159
code.vegaprotocol.io/data-node/api.(*tradingDataService).ObserveEventBus(0xc0001701e0, 0x153f638, 0xc01b4de340, 0x0, 0x0)
	/jenkins/workspace/vegaprotocol_data-node_develop/data-node/api/trading_data.go:2172 +0x37e
code.vegaprotocol.io/protos/data-node/api/v1._TradingDataService_ObserveEventBus_Handler(0x137f540, 0xc0001701e0, 0x153abb8, 0xc01b49af00, 0x1d14778, 0xc01b4db7a0)
	/jenkins/GOPATH/pkg/mod/code.vegaprotocol.io/protos@v0.41.1-0.20210826163644-b2e3ca672bc7/data-node/api/v1/trading_data.pb.go:8726 +0xb0
google.golang.org/grpc.(*Server).processStreamingRPC(0xc000361a40, 0x1540298, 0xc000039200, 0xc01b4db7a0, 0xc0002f0630, 0x1cd2b40, 0x0, 0x0, 0x0)
	/jenkins/GOPATH/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1533 +0xcd8
google.golang.org/grpc.(*Server).handleStream(0xc000361a40, 0x1540298, 0xc000039200, 0xc01b4db7a0, 0x0)
	/jenkins/GOPATH/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1613 +0xca5
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0002943c0, 0xc000361a40, 0x1540298, 0xc000039200, 0xc01b4db7a0)
	/jenkins/GOPATH/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:934 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/jenkins/GOPATH/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:932 +0x1fd
```

--- a/cmd/data-node/node/node_pre.go
+++ b/cmd/data-node/node/node_pre.go
@@ -178,6 +178,12 @@ func (l *NodeCommand) preRun(_ []string) (err error) {
 		}
 	}()
 
+	l.broker, err = broker.New(l.ctx, l.Log, l.conf.Broker)
+	if err != nil {
+		l.Log.Error("unable to initialise broker", logging.Error(err))
+		return err
+	}
+
 	// plugins
 	l.settlePlugin = plugins.NewPositions(l.ctx)
 	l.notaryPlugin = plugins.NewNotary(l.ctx)
@@ -206,12 +212,6 @@ func (l *NodeCommand) preRun(_ []string) (err error) {
 	l.epochService = epochs.NewService(l.Log, l.conf.Epochs, l.epochStore)
 	l.delegationService = delegations.NewService(l.Log, l.conf.Delegations, l.delegationStore)
 	l.nodeService = nodes.NewService(l.Log, l.conf.Nodes, l.nodeStore, l.epochStore)
-
-	l.broker, err = broker.New(l.ctx, l.Log, l.conf.Broker)
-	if err != nil {
-		l.Log.Error("unable to initialise broker", logging.Error(err))
-		return err
-	}
 
 	l.broker.SubscribeBatch(
 		l.marketEventSub, l.transferSub, l.orderSub, l.accountSub,

